pages/domain-gift.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import type { NextPage } from "next";
+import homeStyles from "../styles/Home.module.css";
+import DiscountEndScreen from "../components/discount/discountEndScreen";
+import DiscountCheckoutScreen from "../components/discount/discountCheckoutScreen";
+
+// Create a new discount in utils to create a new discount campaign
+import { domainGift } from "../utils/discounts/domainGift";
```

Comment:
Replace domain gift by freeRegistration everywhere. 

It's a better naming cause for the code it gives more information about you're doing.